Code/GraphMol/AddHs.cpp

```diff
@@ -885,11 +971,17 @@ void removeHs(RWMol &mol, const RemoveHsParameters &ps, bool sanitize) {
     }
   }
   boost::dynamic_bitset<> atomsToRemove{mol.getNumAtoms(), 0};
+
   for (auto atom : mol.atoms()) {
     if (shouldRemoveH(mol, atom, ps)) {
       atomsToRemove.set(atom->getIdx());
     }
   }  // end of the loop over atoms
+
+  // Once we know which H atoms would be removed, filter out those that
+  // would cause any SGroups to become empty
+  filter_sgroup_emptying_hydrogens(mol, atomsToRemove);
```

Comment:
minor, but couldn't this be:
```suggestion
  if (ps.removeInSGroups)  {
    filter_sgroup_emptying_hydrogens(mol, atomsToRemove);
 }
```

Reply:
Sounds reasonable.